Nav improve

ricardoom opened the pull request:
Wrapping up the stylings here. After review and comments on this branch. If there is nothing major that needs doing, I think it should be merged and any further design work can take place on other branches.

Commit:
moved the chapter nav section to its own ul element and pinned it to the bottom of the viewport as per recommendation of @mgiraldo. made some color changes but those are mostly for sanity and will probably be changed

Commit:
added markup for an svg icon. added some aria attributes. update to scss file to style the new icon and adding in mediaqueries to expose a text label when the viewport gets larger

Commit:
raw svg markup for reference

Commit:
styling & layout for the new icon

Commit:
improved the svg icons a bit. added arrow icons to the chapter pagers

Commit:
left & right arrow wedge icons added for reference

Commit:
modified a media a media query

Commit:
Merge branch 'master' into nav-improve

Commit:
improved markup for svg wedge arrow things. aria attributes still need work.

Commit:
added rules for mobile so the chapter navigation arrows at bottom layout nicely, added a Chapters span to so its clear what the arrows do, as the viewport gets bigger that label disappears and labels for each arrow show up. TODO: remove placeholder svg icons w/ more intentional ones. Additional aria attributes for the menus and the navigations. Further testing on smalls screens w/ assistave tech enabled

Commit:
added preserveAspectRatio attribute to the svg icons

aslagle committed:
Hide settings when you click outside it.

aslagle committed:
Show TOC on first load.

aslagle committed:
Added bottom margin to iframe.

Commit:
working in new svg icons to be visually unified

Commit:
created and added unified svg icons. sass to go along

Commit:
Merge branch 'nav-improve' of github.com:NYPL-Simplified/webpub-viewer into nav-improve

Commit:
updates to the settings menu, while its not finally designed the overall layout is much better and should be easier to use on a small handset

Commit:
added a touch of transparency to the settings overlay to see if it helps a user see the type being increased or decreased... so far it does that...

Commit:
messing aroun w/ blue as the main color. exploring button sizes

Commit:
working towards making a smaller control menu. making buttons inline so overall height of the the dialog is about half of an iPhone5's vertical viewport height.

Commit:
added borders to the top bar buttons so they seem more button like

Commit:
properly placing `aria-labelledby`  attriubtes w/ correct referrer
modified some css, but not much

Commit:
working towards a final design

Commit:
removed commented out code, continued working on navigation at small and large screens. adjusted button styles to be a little closer to the simplyE style. TODO: continue styling w/ an eye towards the simplyE controls as per @mgiraldo's suggestions

aslagle commented:
I need to update the tests for a couple of the changes I made on this branch before it's merged.

mgiraldo commented:
screenshots?

aslagle commented:
I updated the demo http://dev-mta.librarysimplified.org/examples/alice/

mgiraldo commented:
maybe have a sample “Aa” in the settings so user knows how much bigger/smaller the font is? Right now the settings cover a good portion of the book... or put a semi-transparent background to the settings?

mgiraldo commented on src/styles/sass/main.scss:
```diff
@@ -6,87 +6,319 @@
   height: calc(100% - 2rem);
   left: 0;
   overflow: scroll;
   position: fixed;
```
add this rule: 

`-webkit-overflow-scrolling:touch`

so that iOS shows the smooth-enabled scrolling… this is iOS only but still we should take advantage of it

ricardoom commented:
![page settings l](https://cloud.githubusercontent.com/assets/20174701/24116634/cfb8ea3c-0d7d-11e7-8618-d1241a6e58f0.png)
![page l](https://cloud.githubusercontent.com/assets/20174701/24116633/cfb8d9a2-0d7d-11e7-82d6-f6cff47f9602.png)
![page settings m](https://cloud.githubusercontent.com/assets/20174701/24116631/cfb703ac-0d7d-11e7-99a6-a0e30848e3d4.png)
![page sm](https://cloud.githubusercontent.com/assets/20174701/24116632/cfb7ea92-0d7d-11e7-9853-c257dd3df2bd.png)
![page settings sm](https://cloud.githubusercontent.com/assets/20174701/24116637/cfbe97e8-0d7d-11e7-88fd-94c121d51649.png)
![contents-m](https://cloud.githubusercontent.com/assets/20174701/24116635/cfb94e0a-0d7d-11e7-8641-4a8851a73f15.png)
![contents-sm](https://cloud.githubusercontent.com/assets/20174701/24116636/cfbca960-0d7d-11e7-9d9d-7fb9c27eea1f.png)

Commit:
modified the language to be shorter, Readoffline gets us the same result as the previous language.

Commit:
removed the word View from the button

Commit:
removed the word view from the scrolling button

Commit:
adusted the width of the read off line button

aslagle committed:
Updated tests.

aslagle commented on src/BookSettings.ts:
```diff
@@ -3,19 +3,19 @@ import * as HTMLUtilities from "./HTMLUtilities";
 
 const optionTemplate = (className: string, label: string) => `
-    <li><a href='#' class='${className}'>${label}</a></li>
+    <li><a href='#' class='${className}' role="button">${label}</a></li>
```
should this be an actual button?

mgiraldo replied:
good idea. yes

aslagle commented on src/IFrameNavigator.ts:
```diff
@@ -13,19 +13,70 @@ import * as HTMLUtilities from "./HTMLUtilities";
+          </a>
+        </li>
+        <li>
+          <a rel="contents" class="disabled" role="button" aria-labelledby="contents" aria-haspopup="true" aria-expanded="false">
```
is aria-expanded supposed to change when the popup is showing?

mgiraldo replied:
yes. should become `true`. [this example](https://nypl.github.io/design-toolkit/sections/buttons.html#menuButton) shows a similar context (with jQuery code)

aslagle commented on src/styles/sass/img/contents.svg:
```diff
@@ -0,0 +1,7 @@
+<svg viewBox="0 0 24 24">
```
why are there svg files as well as the stuff in IFrameNavigator? are they both used?

ricardoom replied:
Ah, no, those are for reference only, they are not used. I put them there for ease of transport from one computer to the next. We can remove them from the project for tidiness.

aslagle replied:
Yeah, let's remove them, unless it's still useful for you to have them here.

Commit:
modified the width of the buttons

Commit:
Merge branch 'nav-improve' of github.com:NYPL-Simplified/webpub-viewer into nav-improve

aslagle committed:
Changed settings links to buttons and added class to li.

aslagle commented:
In the toc, it would be nice if you could click anywhere in a line, and there could be more of a hover state. The toc link with "active" class should also be highlighted somehow - that's the chapter you're currently on.

aslagle committed:
Update aria-expanded on contents and settings links as views are shown/hidden.

Commit:
changed increase & decrese to A & A (little A big A). removed commeted code

Commit:
fixed merg conflict

Commit:
changed Download this bookf offline reading to Read offline

Commit:
making styling changes for new button markup.

Commit:
changed the increase decrease buttons to big a little a

Commit:
modified the `<a>` stylings to accomodate new `<button>` markup

Commit:
added hover state for the chapter menu

Commit:
made a more global rule for removigin ul li list styles
removed a bunch of commented code
moved the the chapter navigation styles to the contents view rule to avoid hovers being activated in unxpected places.

Commit:
changed the background transparency on the menu so there is visual feedback when increasing or decreasing type size
removed default width on the enable offline button
added widht to said button on larger screen size
and a few minor layout changes, mostly just paddings etc...

ricardoom commented:
@aslagle as per your request I added hover states for the TOC. Could you add an `.active` class to the current chapter? Seems like a job for JS...

aslagle commented:
The active class should be there already, unless something broke it...

aslagle committed:
Updated tests.

ricardoom commented on src/IFrameNavigator.ts:
```diff
@@ -13,19 +13,70 @@ import * as HTMLUtilities from "./HTMLUtilities";
+          </a>
+        </li>
+        <li>
+          <a rel="contents" class="disabled" role="button" aria-labelledby="contents" aria-haspopup="true" aria-expanded="false">
```
We should probably change these to `<button>`s as well

ricardoom commented on src/IFrameNavigator.ts:
```diff
@@ -13,19 +13,70 @@ import * as HTMLUtilities from "./HTMLUtilities";
+          </a>
+        </li>
+        <li>
+          <a class="settings" role="button" aria-labelledby="settings-menu" aria-expanded="false" aria-haspopup="true">
```
`<button>`

Commit:
Changed the language somewhat to make a little more sense... this will probably change again, once we have consensus

aslagle committed:
Changed contents and settings links to buttons and cleaned up indentation.

Commit:
added a lighter gray to the mix.

Commit:
fixed up the hover state for the TOC to give the current .active class a hover
removing some margins and paddings to compress the settings menus
added vertical alignment to get the big A little A buttons to layout better

Commit:
Merge branch 'nav-improve' of github.com:NYPL-Simplified/webpub-viewer into nav-improve

Commit:
remvoed `role="button"` from the `<button>`

Commit:
added some sectional comment to the file

Commit:
code consolidation
made more general button rules.

Commit:
fixed minor layout issues

ricardoom commented:
ready for another review. I can post screen shots if need be.
@aslagle can you take a look when you have a moment?

aslagle commented on src/IFrameNavigator.ts:
```diff
@@ -177,13 +234,13 @@ export default class IFrameNavigator implements Navigator {
         let statusMessage = "";
         if (status === CacheStatus.Uncached) {
-            statusMessage = "Not available offline";
+            statusMessage = "Download";
```
This seems weird to me, next to the read offline button. Maybe there shouldn't be an additional message in this case.

ricardoom replied:
I would like to test removing all the labels and see if users have trouble with the controls...  what do you think about that?

aslagle replied:
We can try it, but I think it's good for a user to know whether the book is ready to use offline, vs still downloading or there was an error.

aslagle replied:
I guess the status message could replace the button, and not be there when the button is showing. I don't think we need labels for the view and font size controls.

ricardoom replied:
I've removed the `View` & `Font Size` in my latest commit.

aslagle commented on src/styles/sass/main.scss:
```diff
@@ -6,87 +6,340 @@
   font-family: $sans-serif-stack;
   font-size: $basefont;
-  height: 4.65rem;
+  margin: 0 auto !important;
```
important?

aslagle replied:
if you're overriding https://github.com/NYPL-Simplified/webpub-viewer/blob/master/examples/alice/index.html#L9 just change that line

ricardoom replied:
got it.

ricardoom replied:
just to be clear you're suggesting:
removing  the inline `style` on the `<body>` tag and change the scss to this:
```body {
  font-family: $sans-serif-stack;
  font-size: $basefont;
  margin: 0 auto;
  overflow: hidden;

  @media (min-width: 45rem) {
    width: 40rem;
  }
}

aslagle replied:
👍

aslagle commented:
I found a bug in my margin code - when you go to the next chapter by turning a page the top and bottom margins go away.

aslagle committed:
Fixed margins after switching chapters with links hidden.

Commit:
moved the inline styles to the style sheet.

Commit:
same, inline style on body tag moved to the stylesheet

Commit:
Merge branch 'nav-improve' of github.com:NYPL-Simplified/webpub-viewer into nav-improve

Commit:
removed commmented code

Commit:
removed the `View` & `Font Size` labels

ricardoom commented on src/BookSettings.ts:
```diff
@@ -116,12 +116,12 @@ export default class BookSettings {
                 optionTemplate("", bookView.name, bookView.label)
             );
-            sections.push(sectionTemplate("View", viewOptions.join("")));
+            sections.push(sectionTemplate("", viewOptions.join("")));
```
@aslagle maybe you dont want an empty string here... the compiler was complaining otherwise and I didnt want to muck about too much w/ template logic, so I did it this way to get the job done...

aslagle replied:
If none of them have labels I can get rid of that argument.

ricardoom commented on src/BookSettings.ts:
```diff
@@ -116,12 +116,12 @@ export default class BookSettings {
         if (this.fontSizes.length > 1) {
             const fontSizeOptions = optionTemplate("font-setting", "decrease", "A") + optionTemplate("font-setting", "increase", "A");
-            sections.push(sectionTemplate("Font Size", fontSizeOptions));
+            sections.push(sectionTemplate("", fontSizeOptions));
```
same deal here...

Commit:
cleaning up the style sheets. moving the sections to their own partials, for ease of use

Commit:
removed @warn directive

aslagle committed:
Got rid of section label argument.

aslagle committed:
Took out offline status message when read offline button is showing.

Commit:
more code cleanings and some sectional commentatary

Commit:
Merge branch 'code-cleanup' into nav-improve

Commit:
Merge branch 'nav-improve' of github.com:NYPL-Simplified/webpub-viewer into nav-improve

Commit:
minor futz w/ the comment block

ricardoom commented:
merge?

aslagle committed:
Removed svg files.
